data/modules/goja-usage.ts

```diff
@@ -0,0 +1,1684 @@
+        /**
+         * List stream presences.
+         *
+         * @param stream - Stream object.
+         * @param includeHidden - Optional argument to include hidden presences in the list or not, default true.
+         * @param includeNotHidden - Optional argument to include not hidden presences in the list or not, default true.
+         * @returns List of presence objects.
+         */
+        streamUserList(stream: Stream, includeHidden?: boolean, includeNotHidden?: boolean): Presence[];
+
+        /**
+         * Get presence of user in a stream.
+         *
+         * @param userID - User ID.
+         * @param sessionID - Session ID.
+         * @param stream - Stream data.
+         * @returns Presence object.
+         */
+        streamUserGet(userID: string, sessionID: string, stream: Stream): Presence;
+
+        /**
+         * Add a user to a stream.
+         *
+         * @param userID - User ID.
+         * @param sessionID - Session ID.
+         * @param stream - Stream data.
+         * @param hidden - Opt. If hidden no presence events are generated for the user.
+         * @param persistence - Opt. By default persistence is enabled, if the stream supports it.
+         * @param status - Opt. By default no status is set for the user.
+         */
+        streamUserJoin(userID: string, sessionID: string, stream: Stream, hidden?: boolean, persistence?: boolean, status?: string): void;
+
+        /**
+         * Update user status in a stream.
+         *
+         * @param userID - User ID.
+         * @param sessionID - Session ID.
+         * @param stream - Stream data.
+         * @param hidden - Opt. If hidden no presence events are generated for the user.
+         * @param persistence - Opt. By default persistence is enabled, if the stream supports it.
+         * @param status - Opt. By default no status is set for the user.
+         */
+        streamUserUpdate(userID: string, sessionID: string, stream: Stream, hidden?: boolean, persistence?: boolean, status?: string): void;
+
+        /**
+         * Have a user leave a stream.
+         *
+         * @param userID - User ID.
+         * @param sessionID - Session ID.
+         * @param stream - Stream data.
+         */
+        streamUserLeave(userID: string, sessionID: string, stream: Stream): void;
+
+        /**
+         * Kick user from a stream.
+         *
+         * @param presence - User presence data.
+         * @param stream - Stream data.
+         */
+        streamUserKick(presence: Presence, stream: Stream): void;
+
+        /**
+         * Count the users in a stream.
+         *
+         * @param stream - Stream data.
+         * @returns the number of users in the stream.
+         */
+        streamCount(stream: Stream): number
+
+        /**
+         * Close a stream.
+         *
+         * @param stream - Stream data.
+         * @returns the number of users in the stream.
+         */
+        streamClose(stream: Stream): void;
```

Comment:
Closing a stream removes all presences from it, effectively "deleting" the stream. By design presences formerly on the closed stream don't get any events to notify them of this.

Reply:
I'll add the description we have in the docs: `Closing a stream removes all presences currently on it. It can be useful to explicitly close a stream and enable the server to reclaim resources more quickly.`